### PR TITLE
Bugfix/arch 619

### PR DESCRIPTION
**Pull request by shafi1379**
@afoerster @rk-fsdev  Fixed ARCH-619 bugs, to enable and disable OK button based on dropdown selection and and user entry.

### CODE REVIEW

**Review by afoerster — COMMENTED** [review]
What's the new behavior of the OK button now? I'm not sure I specified it very well in the story.

**Comment by afoerster** [comment]
@shafi1379 so a valid user has to be in the box for the user to hit OK?

**Comment by rk-fsdev** [comment]
@shafi1379 there is a state which changes when user selects a valid user.
`/form/SimpleMemberForm/distinguishedName`
if this state `distinguishedName` is set, then it means user selects a valid user.
Then we can make `OK` button active.
We don't need to add `onChange` props in `MemberSearchBar`.

**Comment by rk-fsdev** [comment]
@shafi1379 we don't need `handleMemberChange`, `setOKButtonProps` and `getOKButtonProps`. You can get `distinguishedName` using a selector in `selectors.ts`, and set it as a prop for `SimpleMemberRequest` modal. Like below:
`okButtonProps = {{disabled: !distinguishedName}}`

File: ui/containers/WorkspaceDetails/index.tsx [hunk 1]
```diff
@@ -97,17 +98,27 @@ interface Props extends RouteComponentProps<DetailsRouteProps> {
...
+  setOKButtonProps: (obj: any) => void;
 }
 
 class WorkspaceDetails extends React.PureComponent<Props> {
   public delayedFetchUsers = throttle((v: string) => {
     this.props.getUserSuggestions(v);
   }, 2000);
 
+  public getOKButtonProps = (v: string) => {
+    const _return = { disabled: true };
```
Comment (rk-fsdev) [inline ui/containers/WorkspaceDetails/index.tsx]: @shafi1379 We are using camelCase naming convention for variables' naming. I'd like you to name variables with camelCase naming convention. I know why you used `_` for the suffix, but it seems better to use camelCase in naming convention.

File: ui/containers/WorkspaceDetails/index.tsx [hunk 2]
```diff
@@ -104,10 +105,17 @@ class WorkspaceDetails extends React.PureComponent<Props> {
     this.props.getUserSuggestions(v);
   }, 2000);
 
+  public getOKButtonProps = (v: string) => {
```
Comment (rk-fsdev) [inline ui/containers/WorkspaceDetails/index.tsx]: Do we need still this method? I don't think it is necessary.

**Review by rk-fsdev — APPROVED** [review]

**Review by afoerster — APPROVED** [review]

File: ui/containers/WorkspaceDetails/index.tsx [hunk 3]
```diff
@@ -339,10 +345,11 @@ class WorkspaceDetails extends React.PureComponent<Props> {
         </Tabs>
         <Modal
           visible={activeModal === ModalType.SimpleMember}
-          title="Add A Member"
+          title={'Add A Member'}
```
Comment (rk-fsdev) [inline ui/containers/WorkspaceDetails/index.tsx:348]: @shafi1379 pls refactor this code. to "Add A Member"

**Pull request merged**